### PR TITLE
#54 integration with backend 1

### PR DESCRIPTION
**Pull request by fatmaalzharaa**

### CODE REVIEW

File: candi-meter/candi-meter-ui/src/app/core/services/config.service.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+@Injectable()
+
+export class ConfigService {
+  private config!: Configuration;
+  private env: Environment = { env: 'development' };
+  configUrl = '/config';
+  envUrl = 'env';
+ // private httpClient:HttpClient;
+  constructor( private httpClient:HttpClient) {
+ //   this.httpClient = new HttpClient(this.httpBackend);
```
Comment (Tasneem-Gad) [inline candi-meter/candi-meter-ui/src/app/core/services/config.service.ts:16]: please don't push with comment

File: candi-meter/candi-meter-ui/src/app/profit.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export interface Profit {
+  id: number;
```
Comment (Tasneem-Gad) [inline candi-meter/candi-meter-ui/src/app/profit.ts:2]: this interface part of our project?

File: candi-meter/candi-meter-ui/src/app/repo.service.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { HttpClient } from '@angular/common/http';
```
Comment (Tasneem-Gad) [inline candi-meter/candi-meter-ui/src/app/repo.service.ts:1]: same

File: candi-meter/candi-meter-ui/.vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+    "cSpell.words": [
```
Comment (Tasneem-Gad) [inline candi-meter/candi-meter-ui/.vscode/settings.json:2]: are you need this file

**Review by Tasneem-Gad — CHANGES_REQUESTED** [review]

File: candi-meter/candi-meter-ui/src/app/core/services/config.service.ts (hunk 1 above)
Comment (fatmaalzharaa) [reply candi-meter/candi-meter-ui/src/app/core/services/config.service.ts]: ok ..
done

File: candi-meter/candi-meter-ui/src/app/profit.ts (hunk 2 above)
Comment (fatmaalzharaa) [reply candi-meter/candi-meter-ui/src/app/profit.ts]: no ,
done

File: candi-meter/candi-meter-ui/src/app/repo.service.ts (hunk 3 above)
Comment (fatmaalzharaa) [reply candi-meter/candi-meter-ui/src/app/repo.service.ts]: done

File: candi-meter/candi-meter-ui/.vscode/settings.json (hunk 4 above)
Comment (fatmaalzharaa) [reply candi-meter/candi-meter-ui/.vscode/settings.json]: no 
done

**Comment by ZinabHashem** [comment]
solve conflict please.

File: candi-meter/candi-meter-ui/src/app/core/services/resource.service.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    return this.httpClient
+      .get<T>(`${this.APIUrl}/${url}?${params.toString()}`)
+      .pipe(
+        map((list) => list),
+        catchError((err) => {
+          throw new Error(err.message);
+        })
+      );
+  }
+  et(id: string | number): Observable<T> {
```
Comment (ZinabHashem) [inline candi-meter/candi-meter-ui/src/app/core/services/resource.service.ts:47]: fix function name

File: candi-meter/candi-meter-ui/src/app/app.component.ts [hunk 6]
```diff
@@ -1,10 +1,13 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
```
Comment (ZinabHashem) [inline candi-meter/candi-meter-ui/src/app/app.component.ts:1]: OnInit why add this without use

**Comment by ZinabHashem** [comment]
solve conflicts

**Pull request merged**